# Optics -- Homework 7
Jarod Boone 

In [1]:
%matplotlib notebook 
import numpy as np # math library 
from bqplot import pyplot as plt # plotting library

## Problem 1 

Laser medium pumped by laser at $\lambda_p = 690 * 10^{-9}\enspace\text{[m]}$ and with a laser frequency at $\lambda_l = 930 * 10^{-9}\enspace\text{[m]}$. Because of the energy diagram and the fact that the pump and laser share the same lower energy state but have different high energy states, I'm going to assume that this is a 3-level laser system.

This means population inversion is possible but requires some amount of input intensity much greater than 0. 

We can calculate the laser frequency ($\frac{c}{\lambda_l}\cdot 2\pi = \omega_l$) in order to determine the saturation intensity $I_{sat} = \frac{\hbar\cdot\omega}{\tau\cdot\sigma}$ at which the laser intensity saturates. 

In [2]:
c = 3e8
h = 6.626e-34

pump_wavelength = 690e-9
laser_wavelength = 930e-9

sigma = 10e-23 
tau = 10e-6
laser_frequency = c/laser_wavelength  

saturation_intensity = (h * laser_frequency)/(sigma * tau)

print(f'saturation intensity = {saturation_intensity:e} [W/m]')

saturation intensity = 2.137419e+08 [W/m]


**a)** Given that the laser is a 3 level system we can solve for the population inversion

$$\Delta N = N \cdot \frac{1 - (I/I_{sat})}{1 + (I/I_{sat})} \implies \Delta N < 0 \quad \textbf{iff} \quad I > I_{sat} $$

Therefore we must set the pump intensity $I_{pump} =$ `2.14 * 10^8 [W/m]` in order to obtain a population inversion 

**b)** If we pump at exactly this intensity there will be zero produced laser intensity because although the population is inverted we do not necessarily have a gain that overcomes the loss due to reflection and absorption within the laser cavity. More precisely: there exists some minimum intensity $I_{threshold} > I_{sat}$ such that there is no lasing if the pump intensity is less than $I_{threshold}$ 

Therfore the intensity of the laser radiation $I_{laser} =$ `0` given that  $I_{pump} = I_{sat}$ `2.14 * 10^8 [W/m]`

**c)** If we raise the pump power to be far past this value then we will have some lasing! This will be proportional to the pump power and increase linearly by the slope efficiency. 

We know that the output intensity is directly porportional to $\Delta N$ so we can therefore say that for a given pump intensity $I$ the output intensity is for some constant $A$

$$ I_{laser} = A \cdot \frac{1 - (I - I_{threshold})/I_{sat}}{1 + (I - I_{threshold})/I_{sat}} = A \cdot (\frac{1}{1 + (I - I_{threshold})/I_{sat}} -  \frac{(I - I_{threshold})/I_{sat}}{1 + (I - I_{threshold})/I_{sat}}) $$

Upon inspection of the equation and isolation of the independent variable we see that this constant $A$ divided by $I_{sat}$ is the slope of $I_{laser}$ which is some non linear function of iput intensity

$$  I_{laser} \frac{A}{I_{sat}} \cdot (\frac{I_{sat}}{1 + (I - I_{threshold})/I_{sat}} -  \frac{(I - I_{threshold})}{1 + (I - I_{threshold})/I_{sat}}) = \frac{A}{I_{sat}} \cdot \frac{I_{sat}(I_{sat} + I_{threshold} - I)}{I_{sat} - I_{threshold} + I} $$

Thus, by definition of the slope efficiency and per the givens in the problem we have $\frac{A}{I_{sat}} = $ `60%`


If we have the condition $ I >> I_{threshold} > I_{sat} $ Then the term $(I - I_{threshold})/I_{sat}$ dominates the first equation and the equation converges to 

$$ I_{laser} = \lim_{I \to \infty} A \cdot \frac{1 - (I - I_{threshold})/I_{sat}}{1 + (I - I_{threshold})/I_{sat}} = A = I_{sat} * 0.6 = 1.28 \cdot 10^{8} \enspace [W/m^2]$$




# Problem 2 

We have a laser that we pump with another laser that emits green light. The threshold pump power of the pump needed to lase $P_{threshold} = 0.4W$

The laser pump has a slope efficiency of 10%. The pump laser is focused to a spot size diameter of $50 \enspace[\mu m ]\implies r_{focus} = 25 \enspace[\mu m] $ inside the gain medium.
The absorption cross-section is given by $\sigma = 4.1 * 10^{-19} \enspace[cm^2] = 4.1 * 10^{-24} \enspace [m^2]$
 at the laser wavelength
of $\lambda_l = 789 \enspace [nm]$. The expiration lifetime of the laser meduim is given as $\tau = 3.2 \mu s$.

We want to deal in power so we use the following equation to convert from saturation intensity to saturation power
$P_{sat} = I_{sat}\cdot A$

We simply need to get a formula for the population inversion to determine a formula for the curve because the output power is porportional to population inversion (gain). We approximate this laser as a 4 level system as most lasers are and we are not given an energy level diagram. More specifically for some input power $P$ and an unknown constant $A$ we have: 

$$ P_{laser} = A \cdot \frac{1 - (P - P_{threshold})/P_{sat}}{1 + (P - P_{threshold})/P_{sat}} $$

At values of $P$ that are near the threshold we have that $P - P_{threshold} \to 0$ and therefore the denominator is dominated by 1 and we have  
$$ P_{laser} = A \cdot (P - P_{threshold})/P_{sat}$$

Now the slope of the line in the near threshold regime is thus $\frac{A}{P_{sat}}$ and we can use the given slope efficiency to solve for $A = P_sat \cdot (0.1)$

In [3]:
# Problem 2 Laser Parameters 
laser_wavelength = 789e-9 
tau = 3.2e-6
laser_frequency = c/laser_wavelength  
sigma = 4.1e-23 # -19 cm^2 ==> -23 m^2
threshold_power = 0.4

# Focus geometry 
r = 25e-6 
focus_area = np.pi * r**2

# Saturation calcuation
saturation_intensity = (h * laser_frequency)/(sigma * tau)
print(f'Saturation Intensity = {saturation_intensity:e}')
saturation_power = focus_area * saturation_intensity
print(f'Saturation Power = {saturation_power:e}')

# Calculate Population Inversion constant 
A = saturation_power * 0.1

def get_output_power(input_power) : 
    if input_power < threshold_power:
        return 0

    dp = (input_power - threshold_power)/saturation_power
    return A * ((dp)/(1 + dp))

# Plot this from zero to ten watts 
domain = np.linspace(0,10,1000)
y = list(map(get_output_power,domain))

plt.figure(title='Output Power vs Input Input Power',interaction=None)

plt.plot(domain,y)
plt.show(display_toolbar=False)

plt.xlabel("Pump Power [W] ")
plt.ylabel("Output Laser Power [W]")
plt.xlim(0,10)

Saturation Intensity = 1.920268e+09
Saturation Power = 3.770437e+00


Figure(axes=[Axis(scale=LinearScale()), Axis(orientation='vertical', scale=LinearScale())], fig_margin={'top':…

LinearScale(max=10.0, min=0.0)

## Problem 3

At or above threshold the power circulating inside a laser cavity is given by 

$$P_{circ} = (\frac{g}{L} - 1) \frac{P_{sat}}{2}$$

$L$ is the net loss in the cavity per round trip. 

$$ L = \delta_{OC} + \delta_{other} $$

Here $\delta_{OC} = 1 - R$ Is the loss due to the output coupling and $\delta_{other}$ is the loss due to all other mechanisms. 

**a)** In addition to this we know that the power emerging from the laser is given as $P_{out} = \delta_{OC}\cdot P_{circ}$. Below we use the laser parameters given by the paper and graph $P_{circ}$ and $P_{out}$ as a function of the output coupling loss



In [4]:
gain = 0.2 
d_other = 0.01 
p_sat = 6.8 

# Function for getting circular power
def pcirc(d_oc) : 
    return ((gain/(d_oc + d_other)) - 1) * (p_sat/2)

# Function for getting output power 
def pout(d_oc) : 
    return pcirc(d_oc) * d_oc

In [5]:
domain = np.linspace(0,1,1000)
y = list(map(pcirc,domain))

plt.figure(title='Circulating Power vs Delta OC',interaction=None)

plt.plot(domain,y)
plt.show(display_toolbar=False)

plt.xlabel("Delta OC")
plt.ylabel("Circulating Power [W]")


Figure(axes=[Axis(scale=LinearScale()), Axis(orientation='vertical', scale=LinearScale())], fig_margin={'top':…

In [6]:
domain = np.linspace(0,1,1000)
y = list(map(pout,domain))

plt.figure(title='Power Output vs Delta OC',interaction=None)

plt.plot(domain,y)
plt.show(display_toolbar=False)

plt.xlabel("Delta OC")
plt.ylabel("Output Power [W]")

Figure(axes=[Axis(scale=LinearScale()), Axis(orientation='vertical', scale=LinearScale())], fig_margin={'top':…

The plots are only really meaningful on the range of $\delta_{OC} \in [0,0.2]$ because outside of this range either $\delta_{OC}$ is invalid or the power is 0 and there is no lasing so the equations are not valid. The maximum for circulating power is about 65 Watts and occurs at $\delta_{OC} = 0$. The maximum value for output power is apparently 0.48 Watts and ocurs around $\delta_{OC} = 0.4$. No they do not happen at the same value of $\delta_{OC}$

**b)** We can just take the expanded equation for output power as follows 

$$P_{out} = \delta_{OC}\cdot P_{circ} = \delta_{OC}\cdot (\frac{g}{\delta_{OC} + \delta_{other}} - 1) \frac{P_{sat}}{2}$$
Taking the derivative of this function in respect to $\delta_{OC}$ and setting it equal to zero (shouts out wolfram alpha) we find that the maximum occurs at $\delta_{OC} =$ `0.035` 

Plugging this in to our equation for $P_{out}$ we obtain the optimum value


In [7]:
print(f'Maximum Output Power = {pout(0.035)}')

Maximum Output Power = 0.40988888888888886


Thus this laser has a maxiumum output power of `410 [mW]` which is approximately what was found in the paper.

## Problem 4

A large number of photons $n(t)$ circulating in a laser cavity we expect that the exponential decay is modeled as 

$$ n(t) = n_0 e^{\frac{-t}{\tau_{c}}} $$

This decay is due to both $\delta_{OC}$ and $\delta_{other}$ and is reflected in the cavity lifetime $\tau_{c}$. After one round trip $t = T_{rt}$ the number of photons remaining is $n_0 \cdot (1 - L)$. For the laser in problem 3 we can compute this (at peak performance) as 

$$e^{\frac{-T_{rt}}{\tau_{c}}} = 1 - (\delta_{OC} + \delta_{other}) = 1 - (0.035 + 0.01) = 0.955 \implies \frac{-T_{rt}}{\tau_{c}} = ln(0.955) = \implies \tau_{c} = \frac{-T_{rt}}{ln(0.955)} = 21.8T_{rt}$$

Therefore the typical photon makes about `22` round trips before it leaves the cavity 

## Problem 5 

**a)** Using the formula for gain parameters we have 

$$ g_1 = (1 - \frac{L}{R_1}) \enspace g_2 = (1 - \frac{L}{R_2}) \implies g_1 = 1 + L \enspace g_2 = 1 - \frac{2L}{3} $$

We find that the bound is such that the product is between 0 and 1

$$ (1 + L)\cdot(1 - \frac{2L}{3}) = 0 \implies 1 + \frac{1}{3}L - \frac{2}{3}L^2 = 0 \implies L = \frac{3}{2} $$
$$ (1 + L)\cdot(1 - \frac{2L}{3}) = 0 \implies 1 + \frac{1}{3}L - \frac{2}{3}L^2 = 1 \implies L = \frac{1}{2} $$

So therefore we requre $L \in [0.5m, 1.5m]$ for stability

**b)** We can use the formula for the waist size at mirror one given in the Lecture 24 slides 

$$ w_1^2 = \frac{L\lambda}{\pi}\sqrt{\frac{g_2}{g_1(1 - g_1g_2)}}$$

For the condition $L = 1$ we have $g_1 = 2$ and $g_2 = \frac{1}{3}$ and therfore 

$$ \frac{\pi\cdot w_1^2}{\lambda} = (1) \cdot \sqrt{\frac{\frac{1}{3}}{2 \cdot(1 - 2 \cdot\frac{1}{3})}} = \frac{l}{\sqrt{2}} $$

Minding that the radius of curvature is $R_1 = -1$ we have 

$$\frac{1}{q} = \frac{1}{R_1} - j\frac{\lambda}{\pi\cdot w_1^2} = -1 -j \cdot \sqrt{2} \implies q_1 = \frac{-1 + j\sqrt{2}}{3} $$